src/coreclr/src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -653,29 +661,63 @@ internal static void UnregisterMemoryLoadChangeNotification(Action notification)
         /// <summary>
         /// Skips zero-initialization of the array if possible.
         /// If T contains object references, the array is always zero-initialized.
+        /// pinned: true means you want to pin this object that you are allocating
+        /// otherwise it's not pinned.
+        /// pinned requires that T does not contain object references.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)] // forced to ensure no perf drop for small memory buffers (hot path)
```

Comment:
Do the additional checks around the pinned argument get removed when the caller uses a const value at the call site / the default argument value, or might the optional parameter end up impacting the perf of existing call sites?

Reply:
if `AggressiveInlining` works, then the pinned should be const-propagated and related blocks omitted. I did not check in debugger, but this seems a very easy case.

---
In reply to: [392758813](https://github.com/dotnet/runtime/pull/33526#discussion_r392758813) [](ancestors = 392758813)